contrib/scripts/start-with-cloud-init.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/sh
+
+# This script can be used to start a vfkit VM
+# with cloud-init configuration using the SSH key provided by the user.
+# It expects SSH_PUB_KEY and SSH_USER to be provided by the user.
+# These values are used to generate a cloud-init user-data file.
+# The HOST_NAME variable can be set by the user to provide
+# metadata (instance id and name) about the VM. If not set, default
+# name of the VM will be set to vfkit-vm.
+# These values are then passed to vfkit using the --cloud-init flag.
+# The $DISK_IMG variable needs to be set by the user to a
+# valid image path for the VM.
+#
+# Once the VM is running, the user can connect to it using their
+# provided key. The VM IP can be found in `/var/db/dhcpd_leases`
+# by searching for the HOST_NAME or MAC address (72:20:43:d4:38:62).
+#
+# Example:
+# $ SSH_USER=test HOST_NAME=vm1 DISK_IMG=Fedora-Cloud-Base-AmazonEC2-41-1.4.aarch64.raw \
+# SSH_PUB_KEY=id_rsa.pub ./contrib/scripts/start-with-cloud-init.sh
+#
+# $ ssh -i id_rsa test@192.168.64.14
+
+set -exu
+
+HOST_NAME=${HOST_NAME:-"vfkit-vm"}
+
+PUBLIC_KEY=$(cat "$SSH_PUB_KEY")
```

Comment:
Maybe you could test that `SSH_PUB_KEY` is set before attempting to read it, and error out if not?